Sources/NIO/Selector.swift

```diff
@@ -614,48 +629,75 @@ final class Selector<R: Registration> {
     ///
     /// After closing the `Selector` it's no longer possible to use it.
     public func close() throws {
+        assert(self.myThread == NIOThread.current)
         guard self.lifecycleState == .open else {
             throw IOError(errnoCode: EBADF, reason: "can't close selector as it's \(self.lifecycleState).")
         }
         self.lifecycleState = .closed
 
         self.registrations.removeAll()
 
-        /* note, we can't close `self.fd` (on macOS) or `self.eventfd` (on Linux) here as that's read unprotectedly and might lead to race conditions. Instead, we abuse ARC to close it for us. */
+        /* note, we can't close `self.fd` (on macOS) or `self.eventFD` (on Linux) here as that's read unprotectedly and might lead to race conditions. Instead, we abuse ARC to close it for us. */
```

Comment:
comment not valid anymore, we do close here 👀

Reply:
thank you, fixed